protocol.go

```diff
@@ -145,7 +145,14 @@ type protocolClient interface {
 	// been populated by WriteRequestHeader. When constructing a stream for a
 	// unary call, implementations may assume that the Sender's Send and Close
 	// methods return before the Receiver's Receive or Close methods are called.
-	NewConn(context.Context, Spec, http.Header) StreamingClientConn
+	NewConn(context.Context, Spec, http.Header) streamingClientConn
+}
+
+// streamingClientConn extends StreamingClientConn with a method for registering
+// a hook when the HTTP request is actually sent.
+type streamingClientConn interface {
+	StreamingClientConn
+	onRequestSend(fn func(*http.Request))
```

Comment:
style guide nit: blank line between embedded and named fields, please :)

Reply:
Done